### PR TITLE
Orsp 48 change answers

### PR DESCRIPTION
**Pull request by juaniorbegozo**
## Addresses
https://broadinstitute.atlassian.net/browse/ORSP-48

## Changes
Include a high level summary of the changes and what they are intended for.

## Testing
Describe for the reviewer how to test the specifics of this PR, both positive and negative cases.

---

- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Get a thumb from Belatrix
- [ ] **Submitter**: Get a thumb from ORSP representative
- [ ] **Submitter**: Merge to develop using github's "Squash and Merge" feature
- [ ] **Submitter**: Delete branch after merge

### CODE REVIEW

File: grails-app/services/org/broadinstitute/orsp/StorageProviderService.groovy [hunk 1]
```diff
@@ -1,27 +1,22 @@
...
 import com.google.api.client.googleapis.auth.oauth2.GoogleCredential
-import com.google.api.client.http.GenericUrl
-import com.google.api.client.http.HttpContent
-import com.google.api.client.http.HttpRequest
-import com.google.api.client.http.HttpRequestFactory
-import com.google.api.client.http.HttpResponse
-import com.google.api.client.http.HttpStatusCodes
-import com.google.api.client.http.HttpTransport
-import com.google.api.client.http.InputStreamContent
+import com.google.api.client.http.*
```
Comment (rushtong) [inline grails-app/services/org/broadinstitute/orsp/StorageProviderService.groovy]: Avoid * imports

File: grails-app/services/org/broadinstitute/orsp/IssueService.groovy [hunk 2]
```diff
@@ -243,6 +281,89 @@ class IssueService implements UserInfo {
...
+     * Handles the update of project key to an existing issue.
+     *
+     * @param issue The existing issue to update.
+     * @param input Map of form arguments that come from any of the various update forms
+     * @return Persisted issue
+     */
+    @Transactional
+    Issue updateProjectkey(Issue issue, Map<String, Object> input) throws DomainException {
+        // update projectKey and type if response to determination questions changed
+            def issueType = IssueType.valueOfPrefix(input.get("type"))
```
Comment (rushtong) [inline grails-app/services/org/broadinstitute/orsp/IssueService.groovy]: Prefer fully typed objects over anonymous `def`s. Similar feedback throughout this method.

File: grails-app/services/org/broadinstitute/orsp/StorageProviderService.groovy [hunk 3]
```diff
@@ -455,4 +451,41 @@ class StorageProviderService implements Status {
...
+        credentials
+    }
+
+    void setCredentials(GoogleCredentials credentials) {
+        this.credentials = credentials
+    }
+
+    void renameStorageDocument(StorageDocument document, String newProjectkey) {
+        String bucket = storageConfiguration.bucket.replace("/", "")
+        Storage storage = StorageOptions.newBuilder().setCredentials(getCredentials()).build().getService()
```
Comment (rushtong) [inline grails-app/services/org/broadinstitute/orsp/StorageProviderService.groovy:481]: Nice that you're using the newer, standardized `Storage` classes.

File: src/main/webapp/projectReview/ProjectReview.js [hunk 4]
```diff
@@ -375,14 +382,25 @@ const ProjectReview = hh(class ProjectReview extends Component {
...
+        this.verifyProjectkeyChanged(project.type);
       }).catch(error => {
       this.props.hideSpinner();
       this.setState(() => { throw error; });
-      });
+      })
   };
 
+  verifyProjectkeyChanged(type) {
+    let projectKey = this.props.projectKey.split('-');
```
Comment (rushtong) [inline src/main/webapp/projectReview/ProjectReview.js]: Can this be a `const`?

File: src/main/webapp/util/DeterminationQuestions.js [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+  else if (projectType === DETERMINATION.NHSR) {
+    type = 'NHSR';
+  }
+  else if (projectType === DETERMINATION.IRB) {
+    type = 'IRB';
+  }
+  else if (projectType === DETERMINATION.EX) {
+    type = 'EX';
+  }
+  return type;
```
Comment (rushtong) [inline src/main/webapp/util/DeterminationQuestions.js]: This block of ifs might be more legible as a switch block.

**Review by rushtong — COMMENTED** [review]
Had a few moments to look. Please take my comments with a grain of salt.

**Comment by juaniorbegozo** [comment]
> Had a few moments to look. Please take my comments with a grain of salt.

Thank you for taking the time. Will take a look at suggestions.

**Review by spatilbroad — APPROVED** [review]
👍

**Pull request merged**